rest/utilities_testing.go

```diff
@@ -2285,3 +2287,72 @@ func (rt *RestTester) getCollectionsForBLIP() []string {
 	}
 	return collections
 }
+
+// Reads continuous changes feed response into slice of ChangeEntry
+func (rt *RestTester) ReadContinuousChanges(response *TestResponse) ([]db.ChangeEntry, error) {
+	var change db.ChangeEntry
+	changes := make([]db.ChangeEntry, 0)
+	reader := bufio.NewReader(response.Body)
+	for {
+		entry, readError := reader.ReadBytes('\n')
+		if readError == io.EOF {
+			// done
+			break
+		}
+		if readError != nil {
+			// unexpected read error
+			return changes, readError
+		}
+		entry = bytes.TrimSpace(entry)
+		if len(entry) > 0 {
+			err := base.JSONUnmarshal(entry, &change)
+			if err != nil {
+				return changes, err
+			}
+			changes = append(changes, change)
+			log.Printf("Got change ==> %v", change)
+		}
+
+	}
+	return changes, nil
+}
+
+// GetNextContinousChange reads the next changes entry from the feed.
+func GetNextContinuousChange(reader *bufio.Reader) (*db.ChangeEntry, error) {
+	var change *db.ChangeEntry
+	for {
+		entry, readError := reader.ReadBytes('\n')
+		if readError != nil {
+			return nil, readError
+		}
+		entry = bytes.TrimSpace(entry)
+		if len(entry) > 0 {
+			err := base.JSONUnmarshal(entry, &change)
+			if err != nil {
+				return nil, err
+			}
+			log.Printf("Got change ==> %v", change)
+			return change, nil
+		}
+	}
+}
+
+// RequireContinuousFeedChangesCount Calls a changes feed on every collection and asserts that the nth expected change is
+// the number of changes for the nth collection.
+func (rt *RestTester) RequireContinuousFeedChangesCount(t testing.TB, username string, numKeyspaces int, expectedChanges []int, timeout int) {
```

Comment:
I think I didn't make this as clear, but I actually think that writing this assertion multiple times for:

`RequireContinousFeedChangesCount(t, username, keyspace string, expected changes)`

Reply:
https://github.com/couchbase/sync_gateway/pull/6073/commits/40ad8fa145d1006e7227564a278b1b60341e5654